sklearn/cluster/tests/test_affinity_propagation.py

```diff
@@ -66,7 +67,14 @@ def test_affinity_propagation_predict():
     labels2 = af.predict(X)
     assert_array_equal(labels, labels2)
 
-
+    
+def test_affinity_propagation_convergence_warning():
+    # Test warning in AffinityPropagation.fit
+    af = AffinityPropagation(affinity="euclidean", max_iter=1)
+    assert_warns(ConvergenceWarning, af.fit, X)
+    assert_raises(ValueError, af.predict, X)
+    
```

Comment:
well the test is deterministic. having entirely magic values is probably not that helpful. If the behavior can be produced with a range of values, picking something in that range seems fine to me.